python-avd/pyavd/_eos_designs/shared_utils/wan.py

```diff
@@ -83,25 +83,68 @@ def wan_interfaces(self: SharedUtils) -> EosDesigns._DynamicKeys.DynamicNodeType
         wan_interfaces = EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3Interfaces(
             [interface for interface in self.l3_interfaces if interface.wan_carrier]
         )
+        # TODO: We could have a scenario where there may be no L3 interfaces configured and one or more
+        # interfaces under L3 Port-Channels might be configured as wan-interfaces.
+        # In such cases, below error may not be accurate.
+        # Also need to avoid potential recursion
+        # i.e. avoid logic in self.wan_interfaces referencing self.wan_port_channels and vice-versa
         if not wan_interfaces:
             msg = "At least one WAN interface must be configured on a WAN router. Add WAN interfaces under `l3_interfaces` node setting with `wan_carrier` set."
             raise AristaAvdError(msg)
```

Comment:
We should probably move this check to somewhere else where we real both types and raise an error if none of them contain any interfaces.

Reply:
Ack - check for both property has been moved to wan_local_carriers(). We raise exception there.